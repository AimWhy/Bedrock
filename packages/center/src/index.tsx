import { mergeBreakpoints } from "@bedrock-layout/spacing-constants";
import PropTypes from "prop-types";
import styled from "styled-components";

export interface CenterProps {
  maxWidth?: number | string;
  centerText?: boolean;
  centerChildren?: boolean;
}

function getSafeMaxWidth(
  breakPoints: Record<string, unknown>,
  maxWidth?: number | string
) {
  if (typeof maxWidth === "string") {
    return maxWidth;
  }

  return typeof maxWidth === "number"
    ? `${maxWidth}px`
    : mergeBreakpoints(breakPoints).medium + "px";
}

export const Center = styled.div.attrs<CenterProps>(() => {
  return {
    "data-bedrock-layout-center": "",
  };
})<CenterProps>`
  @property --maxWidth {
    syntax: "<length>";
    inherits: false;
    initial-value: 1023px;
  }
  --maxWidth: ${({ maxWidth, theme: { breakPoints } }) =>
    getSafeMaxWidth(breakPoints, maxWidth)};

  box-sizing: content-box;

  margin-inline-start: auto;
  margin-inline-end: auto;
  margin-inline: auto;

  max-inline-size: 1023px;

  @supports (
    max-inline-size:
      ${({ maxWidth, theme: { breakPoints } }) =>
        getSafeMaxWidth(breakPoints, maxWidth)}
  ) {
    max-inline-size: var(--maxWidth, 1023px);
  }

  ${(props) => (props.centerText ? "text-align: center;" : "")}

  ${(props) =>
    props.centerChildren
      ? `
  display: flex;
  flex-direction: column;
  align-items: center;
  `
      : ""}
`;

Center.displayName = "Center";

Center.propTypes = {
  maxWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  centerText: PropTypes.bool,
};
