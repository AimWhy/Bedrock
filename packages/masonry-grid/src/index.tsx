import { Grid, GridProps } from "@bedrock-layout/grid";
import {
  Gutter,
  getSafeGutter,
  useTheme,
} from "@bedrock-layout/spacing-constants";
import { forwardRefWithAs } from "@bedrock-layout/type-utils";
import { useResizeObserver } from "@bedrock-layout/use-resize-observer";
import React, {
  CSSProperties,
  Children,
  ComponentPropsWithoutRef,
  cloneElement,
  forwardRef,
  useState,
} from "react";

//Logic forked from is-in-browser npm package
/* c8 ignore next */
const isBrowser =
  typeof window === "object" &&
  typeof document === "object" &&
  (document as Document).nodeType === 9;

const RowSpanner = forwardRef<HTMLDivElement, ComponentPropsWithoutRef<"div">>(
  function RowSpanner({ style, ...props }, ref) {
    const safeStyle = style ?? {};
    return (
      <div
        ref={ref}
        {...props}
        style={{ ...safeStyle, gridRow: "span var(--rows, 1)" }}
      />
    );
  },
);

function Resizer({
  children,
  gutter,
}: React.PropsWithChildren<{ gutter?: Gutter }>) {
  const [rowSpan, setRowSpan] = useState(1);

  const theme = useTheme();

  /* c8 ignore next */
  const childRef = useResizeObserver<HTMLDivElement>(({ target }) => {
    setRowSpan(1);
    const gapString = getSafeGutter(theme, gutter) ?? "1px";

    const maybeGap = isBrowser ? toPX(gapString, target) : null;

    const gap: number = Math.max(maybeGap ?? 1, 1);

    const [child] = Array.from(target.children);
    const height = 1 + Math.min(target.scrollHeight, child.scrollHeight);

    const rowHeight = Math.ceil(height / gap);

    setRowSpan(rowHeight);
  });

  return (
    <RowSpanner style={{ "--rows": rowSpan } as CSSProperties} ref={childRef}>
      {Children.map(children, (child) => {
        return cloneElement(child as React.ReactElement, {
          style: {
            display: "block",
            height: "100%",
          },
        });
      })}
    </RowSpanner>
  );
}

/**
 * Props for MasonryGrid component.
 */
export type MasonryGridProps = GridProps;

/**
 * The `MasonryGrid` component is almost identical to the
 * [Grid component](https://www.bedrock-layout.dev/?path=/docs/spacer-components-masonrygrid--docs)
 * except that each item's vertical space will grow independent of each other.
 * The `MasonryGrid` component will then optimize the number of
 * columns based on the `minItemWidth` prop value passed in.
 *
 * `MasonryGrid` does not create standard rows. Instead, it will
 * optimize for the most dense vertical layout that it can achieve based on
 * the space available.
 */
export const MasonryGrid = forwardRefWithAs<"div", MasonryGridProps>(
  function MasonryGrid({ children, style, ...props }, ref) {
    const safeStyle = style ?? {};
    return (
      <Grid
        ref={ref}
        data-bedrock-masonry-grid
        {...props}
        style={{ ...safeStyle, gridTemplateRows: "1px" }}
      >
        {Children.map(children, (child) => (
          <Resizer gutter={props.gutter}>{child}</Resizer>
        ))}
      </Grid>
    );
  },
);

/**
 * This module is adapted from https://github.com/mikolalysenko/to-px/blob/master/browser.js
 */

const PIXELS_PER_INCH = 96;

/* c8 ignore start */
function parseUnit(str: string): [number, string] {
  str = String(str);
  const num = parseFloat(str);

  const [, unit] = str.match(/[\d.\-+]*\s*(.*)/) ?? ["", ""];

  return [num, unit];
}

function toPX(str: string, element?: Element): number | null {
  if (!str) return null;

  const elementOrBody = element ?? document.body;
  const safeStr = (str ?? "px").trim().toLowerCase();

  switch (safeStr) {
    case "vmin":
    case "vmax":
    case "vh":
    case "vw":
    case "%":
      return null;
    case "ch":
    case "ex":
      return getSizeBrutal(safeStr, elementOrBody);
    case "em":
      return getPropertyInPX(elementOrBody, "font-size");
    case "rem":
      return getPropertyInPX(document.body, "font-size");
    case "in":
      return PIXELS_PER_INCH;
    case "cm":
      return PIXELS_PER_INCH / 2.54;
    case "mm":
      return PIXELS_PER_INCH / 25.4;
    case "pt":
      return PIXELS_PER_INCH / 72;
    case "pc":
      return PIXELS_PER_INCH / 6;
    case "px":
      return 1;
    default: {
      const [value, units] = parseUnit(safeStr);

      if (isNaN(value)) return null;

      if (!units) return value;

      const px = toPX(units, element);
      return typeof px === "number" ? value * px : null;
    }
  }
}

/* c8 ignore next */
function getPropertyInPX(element: Element, prop: string): number {
  const [value, units] = parseUnit(
    getComputedStyle(element).getPropertyValue(prop),
  );
  return value * (toPX(units, element) ?? 1);
}

function getSizeBrutal(unit: string, element: Element) {
  const testDIV = document.createElement("div");
  testDIV.style["height"] = "128" + unit;
  element.appendChild(testDIV);
  const size = getPropertyInPX(testDIV, "height") / 128;
  element.removeChild(testDIV);
  return size;
}
/* c8 ignore end */
