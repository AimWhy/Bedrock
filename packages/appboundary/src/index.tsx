import { Center } from "@bedrock-layout/center";
import { mergeBreakpoints } from "@bedrock-layout/spacing-constants";
import React from "react";
import styled, { ThemeContext } from "styled-components";

const safeTheme = { breakPoints: {} };

export const AppBoundary = styled.div.attrs((props) => {
  const { breakPoints } = React.useContext(ThemeContext) ?? safeTheme;
  const maxWidth = mergeBreakpoints(breakPoints).xxlarge;
  return {
    "data-bedrock-layout-appboundary": "",
    children: <Center maxWidth={maxWidth}>{props.children}</Center>,
  };
})`
  padding: 0;
  max-width: 100%;
  overflow: hidden;
  height: 100%;
`;

AppBoundary.displayName = "AppBoundary";
