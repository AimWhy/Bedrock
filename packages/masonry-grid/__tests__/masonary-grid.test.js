import { spacing } from "@bedrock-layout/spacing-constants";
import React from "react";
import { create } from "react-test-renderer";
import { ThemeProvider } from "styled-components";

import { MasonryGrid } from "../src";

describe("MasonryGrid", () => {
  describe("correct usage", () => {
    test("MasonryGrid is not null", () => {
      expect(MasonryGrid).toBeTruthy();
    });

    it("renders all the gutter options", () => {
      Object.keys(spacing).forEach((gutter) => {
        const masonryGrid = create(
          <MasonryGrid gutter={gutter}>
            <div>1</div>
            <div>1</div>
            <div>1</div>
          </MasonryGrid>
        );

        expect(masonryGrid.toJSON()).toMatchSnapshot();
      });
    });

    it("renders custom minItemWidth", () => {
      const masonryGrid = create(
        <MasonryGrid gutter="lg" minItemWidth={320}>
          <div>1</div>
          <div>1</div>
          <div>1</div>
        </MasonryGrid>
      );
      expect(masonryGrid.toJSON()).toMatchSnapshot();
    });

    it("renders custom minItemWidth as string", () => {
      CSS.supports.mockImplementation(() => true);
      const masonryGrid = create(
        <MasonryGrid gutter="lg" minItemWidth="32rem">
          <div>1</div>
          <div>1</div>
          <div>1</div>
        </MasonryGrid>
      );
      expect(masonryGrid.toJSON()).toMatchSnapshot();
    });

    it("renders with theme overrides", () => {
      const masonryGrid = create(
        <ThemeProvider
          theme={{
            breakPoints: { smallOnly: 320 },
            spacing: { "1x": "200px" },
          }}
        >
          <MasonryGrid gutter="1x">
            <div>1</div>
            <div>1</div>
            <div>1</div>
          </MasonryGrid>
        </ThemeProvider>
      );
      expect(masonryGrid.toJSON()).toMatchSnapshot();
    });
  });

  describe("incorrect usage", () => {
    beforeEach(() => {
      jest.spyOn(console, "error");
      console.error.mockImplementation(() => undefined);
    });
    afterEach(() => {
      console.error.mockRestore();
    });

    it("renders default with console error with no gutter input", () => {
      expect(console.error).not.toBeCalled();
      const errorStack = create(
        <MasonryGrid>
          <div>1</div>
          <div>1</div>
          <div>1</div>
        </MasonryGrid>
      );
      expect(console.error).toBeCalled();
      expect(errorStack.toJSON()).toMatchSnapshot();
    });

    it("renders default with wrong gutter input", () => {
      expect(console.error).not.toBeCalled();
      const errorStack = create(
        <MasonryGrid gutter="incorrect">
          <div>1</div>
          <div>1</div>
          <div>1</div>
        </MasonryGrid>
      );

      expect(errorStack.toJSON()).toMatchSnapshot();
    });

    it("renders default with console error with minItemWidth input", () => {
      expect(console.error).not.toBeCalled();
      const errorStack = create(
        <MasonryGrid gutter="lg" minItemWidth={{ value: "incorrect" }}>
          <div>1</div>
          <div>1</div>
          <div>1</div>
        </MasonryGrid>
      );
      expect(console.error).toBeCalled();
      expect(errorStack.toJSON()).toMatchSnapshot();
    });
  });
});
