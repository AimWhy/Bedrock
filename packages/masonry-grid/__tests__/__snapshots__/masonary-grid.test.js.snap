// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MasonryGrid correct usage renders all the gutter options 1`] = `
.c0 {
  box-sizing: border-box;
  --gutter: 0px;
  --minItemWidth: 639px;
  display: grid;
  gap: var(--gutter);
  grid-template-columns: repeat( auto-fit, minmax(min(639px,100%),1fr) );
}

.c0 > * {
  margin: 0;
}

.c2 {
  --rows: 1;
  grid-row: span var(--rows);
}

.c2 > * {
  display: block;
  height: 100%;
}

.c1 {
  grid-template-rows: 1px;
}

@supports (width: 639px ) {
  .c0 {
    grid-template-columns: repeat( auto-fit, minmax(min(var(--minItemWidth),100%),1fr) );
  }
}

<div
  className="c0 c1"
  data-bedrock-layout-masonry-grid=""
>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
</div>
`;

exports[`MasonryGrid correct usage renders all the gutter options 2`] = `
.c0 {
  box-sizing: border-box;
  --gutter: 0.0625rem;
  --minItemWidth: 639px;
  display: grid;
  gap: var(--gutter);
  grid-template-columns: repeat( auto-fit, minmax(min(639px,100%),1fr) );
}

.c0 > * {
  margin: 0;
}

.c2 {
  --rows: 1;
  grid-row: span var(--rows);
}

.c2 > * {
  display: block;
  height: 100%;
}

.c1 {
  grid-template-rows: 1px;
}

@supports (width: 639px ) {

}

@supports (width: 639px ) {
  .c0 {
    grid-template-columns: repeat( auto-fit, minmax(min(var(--minItemWidth),100%),1fr) );
  }
}

<div
  className="c0 c1"
  data-bedrock-layout-masonry-grid=""
>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
</div>
`;

exports[`MasonryGrid correct usage renders all the gutter options 3`] = `
.c0 {
  box-sizing: border-box;
  --gutter: 0.125rem;
  --minItemWidth: 639px;
  display: grid;
  gap: var(--gutter);
  grid-template-columns: repeat( auto-fit, minmax(min(639px,100%),1fr) );
}

.c0 > * {
  margin: 0;
}

.c2 {
  --rows: 1;
  grid-row: span var(--rows);
}

.c2 > * {
  display: block;
  height: 100%;
}

.c1 {
  grid-template-rows: 1px;
}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {
  .c0 {
    grid-template-columns: repeat( auto-fit, minmax(min(var(--minItemWidth),100%),1fr) );
  }
}

<div
  className="c0 c1"
  data-bedrock-layout-masonry-grid=""
>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
</div>
`;

exports[`MasonryGrid correct usage renders all the gutter options 4`] = `
.c0 {
  box-sizing: border-box;
  --gutter: 0.25rem;
  --minItemWidth: 639px;
  display: grid;
  gap: var(--gutter);
  grid-template-columns: repeat( auto-fit, minmax(min(639px,100%),1fr) );
}

.c0 > * {
  margin: 0;
}

.c2 {
  --rows: 1;
  grid-row: span var(--rows);
}

.c2 > * {
  display: block;
  height: 100%;
}

.c1 {
  grid-template-rows: 1px;
}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {
  .c0 {
    grid-template-columns: repeat( auto-fit, minmax(min(var(--minItemWidth),100%),1fr) );
  }
}

<div
  className="c0 c1"
  data-bedrock-layout-masonry-grid=""
>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
</div>
`;

exports[`MasonryGrid correct usage renders all the gutter options 5`] = `
.c0 {
  box-sizing: border-box;
  --gutter: 0.5rem;
  --minItemWidth: 639px;
  display: grid;
  gap: var(--gutter);
  grid-template-columns: repeat( auto-fit, minmax(min(639px,100%),1fr) );
}

.c0 > * {
  margin: 0;
}

.c2 {
  --rows: 1;
  grid-row: span var(--rows);
}

.c2 > * {
  display: block;
  height: 100%;
}

.c1 {
  grid-template-rows: 1px;
}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {
  .c0 {
    grid-template-columns: repeat( auto-fit, minmax(min(var(--minItemWidth),100%),1fr) );
  }
}

<div
  className="c0 c1"
  data-bedrock-layout-masonry-grid=""
>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
</div>
`;

exports[`MasonryGrid correct usage renders all the gutter options 6`] = `
.c0 {
  box-sizing: border-box;
  --gutter: 0.75rem;
  --minItemWidth: 639px;
  display: grid;
  gap: var(--gutter);
  grid-template-columns: repeat( auto-fit, minmax(min(639px,100%),1fr) );
}

.c0 > * {
  margin: 0;
}

.c2 {
  --rows: 1;
  grid-row: span var(--rows);
}

.c2 > * {
  display: block;
  height: 100%;
}

.c1 {
  grid-template-rows: 1px;
}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {
  .c0 {
    grid-template-columns: repeat( auto-fit, minmax(min(var(--minItemWidth),100%),1fr) );
  }
}

<div
  className="c0 c1"
  data-bedrock-layout-masonry-grid=""
>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
</div>
`;

exports[`MasonryGrid correct usage renders all the gutter options 7`] = `
.c0 {
  box-sizing: border-box;
  --gutter: 1rem;
  --minItemWidth: 639px;
  display: grid;
  gap: var(--gutter);
  grid-template-columns: repeat( auto-fit, minmax(min(639px,100%),1fr) );
}

.c0 > * {
  margin: 0;
}

.c2 {
  --rows: 1;
  grid-row: span var(--rows);
}

.c2 > * {
  display: block;
  height: 100%;
}

.c1 {
  grid-template-rows: 1px;
}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {
  .c0 {
    grid-template-columns: repeat( auto-fit, minmax(min(var(--minItemWidth),100%),1fr) );
  }
}

<div
  className="c0 c1"
  data-bedrock-layout-masonry-grid=""
>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
</div>
`;

exports[`MasonryGrid correct usage renders all the gutter options 8`] = `
.c0 {
  box-sizing: border-box;
  --gutter: 1.5rem;
  --minItemWidth: 639px;
  display: grid;
  gap: var(--gutter);
  grid-template-columns: repeat( auto-fit, minmax(min(639px,100%),1fr) );
}

.c0 > * {
  margin: 0;
}

.c2 {
  --rows: 1;
  grid-row: span var(--rows);
}

.c2 > * {
  display: block;
  height: 100%;
}

.c1 {
  grid-template-rows: 1px;
}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {
  .c0 {
    grid-template-columns: repeat( auto-fit, minmax(min(var(--minItemWidth),100%),1fr) );
  }
}

<div
  className="c0 c1"
  data-bedrock-layout-masonry-grid=""
>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
</div>
`;

exports[`MasonryGrid correct usage renders all the gutter options 9`] = `
.c0 {
  box-sizing: border-box;
  --gutter: 2rem;
  --minItemWidth: 639px;
  display: grid;
  gap: var(--gutter);
  grid-template-columns: repeat( auto-fit, minmax(min(639px,100%),1fr) );
}

.c0 > * {
  margin: 0;
}

.c2 {
  --rows: 1;
  grid-row: span var(--rows);
}

.c2 > * {
  display: block;
  height: 100%;
}

.c1 {
  grid-template-rows: 1px;
}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {
  .c0 {
    grid-template-columns: repeat( auto-fit, minmax(min(var(--minItemWidth),100%),1fr) );
  }
}

<div
  className="c0 c1"
  data-bedrock-layout-masonry-grid=""
>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
</div>
`;

exports[`MasonryGrid correct usage renders all the gutter options 10`] = `
.c0 {
  box-sizing: border-box;
  --gutter: 3rem;
  --minItemWidth: 639px;
  display: grid;
  gap: var(--gutter);
  grid-template-columns: repeat( auto-fit, minmax(min(639px,100%),1fr) );
}

.c0 > * {
  margin: 0;
}

.c2 {
  --rows: 1;
  grid-row: span var(--rows);
}

.c2 > * {
  display: block;
  height: 100%;
}

.c1 {
  grid-template-rows: 1px;
}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {
  .c0 {
    grid-template-columns: repeat( auto-fit, minmax(min(var(--minItemWidth),100%),1fr) );
  }
}

<div
  className="c0 c1"
  data-bedrock-layout-masonry-grid=""
>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
</div>
`;

exports[`MasonryGrid correct usage renders all the gutter options 11`] = `
.c0 {
  box-sizing: border-box;
  --gutter: 4rem;
  --minItemWidth: 639px;
  display: grid;
  gap: var(--gutter);
  grid-template-columns: repeat( auto-fit, minmax(min(639px,100%),1fr) );
}

.c0 > * {
  margin: 0;
}

.c2 {
  --rows: 1;
  grid-row: span var(--rows);
}

.c2 > * {
  display: block;
  height: 100%;
}

.c1 {
  grid-template-rows: 1px;
}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {

}

@supports (width: 639px ) {
  .c0 {
    grid-template-columns: repeat( auto-fit, minmax(min(var(--minItemWidth),100%),1fr) );
  }
}

<div
  className="c0 c1"
  data-bedrock-layout-masonry-grid=""
>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
</div>
`;

exports[`MasonryGrid correct usage renders custom minItemWidth 1`] = `
.c0 {
  box-sizing: border-box;
  --gutter: 1rem;
  --minItemWidth: 320px;
  display: grid;
  gap: var(--gutter);
  grid-template-columns: repeat( auto-fit, minmax(min(639px,100%),1fr) );
}

.c0 > * {
  margin: 0;
}

.c2 {
  --rows: 1;
  grid-row: span var(--rows);
}

.c2 > * {
  display: block;
  height: 100%;
}

.c1 {
  grid-template-rows: 1px;
}

@supports (width: 320px ) {
  .c0 {
    grid-template-columns: repeat( auto-fit, minmax(min(var(--minItemWidth),100%),1fr) );
  }
}

<div
  className="c0 c1"
  data-bedrock-layout-masonry-grid=""
>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
</div>
`;

exports[`MasonryGrid correct usage renders custom minItemWidth as string 1`] = `
.c0 {
  box-sizing: border-box;
  --gutter: 1rem;
  --minItemWidth: 32rem;
  display: grid;
  gap: var(--gutter);
  grid-template-columns: repeat( auto-fit, minmax(min(639px,100%),1fr) );
}

.c0 > * {
  margin: 0;
}

.c2 {
  --rows: 1;
  grid-row: span var(--rows);
}

.c2 > * {
  display: block;
  height: 100%;
}

.c1 {
  grid-template-rows: 1px;
}

@supports (width: 32rem ) {
  .c0 {
    grid-template-columns: repeat( auto-fit, minmax(min(var(--minItemWidth),100%),1fr) );
  }
}

<div
  className="c0 c1"
  data-bedrock-layout-masonry-grid=""
>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
</div>
`;

exports[`MasonryGrid correct usage renders with theme overrides 1`] = `
.c0 {
  box-sizing: border-box;
  --gutter: 200px;
  --minItemWidth: 639px;
  display: grid;
  gap: var(--gutter);
  grid-template-columns: repeat( auto-fit, minmax(min(639px,100%),1fr) );
}

.c0 > * {
  margin: 0;
}

.c2 {
  --rows: 1;
  grid-row: span var(--rows);
}

.c2 > * {
  display: block;
  height: 100%;
}

.c1 {
  grid-template-rows: 1px;
}

@supports (width: 639px ) {
  .c0 {
    grid-template-columns: repeat( auto-fit, minmax(min(var(--minItemWidth),100%),1fr) );
  }
}

<div
  className="c0 c1"
  data-bedrock-layout-masonry-grid=""
>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
</div>
`;

exports[`MasonryGrid incorrect usage renders default with console error with minItemWidth input 1`] = `
.c0 {
  box-sizing: border-box;
  --gutter: 1rem;
  --minItemWidth: 639px;
  display: grid;
  gap: var(--gutter);
  grid-template-columns: repeat( auto-fit, minmax(min(639px,100%),1fr) );
}

.c0 > * {
  margin: 0;
}

.c2 {
  --rows: 1;
  grid-row: span var(--rows);
}

.c2 > * {
  display: block;
  height: 100%;
}

.c1 {
  grid-template-rows: 1px;
}

@supports (width: 639px ) {
  .c0 {
    grid-template-columns: repeat( auto-fit, minmax(min(var(--minItemWidth),100%),1fr) );
  }
}

<div
  className="c0 c1"
  data-bedrock-layout-masonry-grid=""
>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
</div>
`;

exports[`MasonryGrid incorrect usage renders default with console error with no gutter input 1`] = `
.c0 {
  box-sizing: border-box;
  --gutter: 0px;
  --minItemWidth: 639px;
  display: grid;
  gap: var(--gutter);
  grid-template-columns: repeat( auto-fit, minmax(min(639px,100%),1fr) );
}

.c0 > * {
  margin: 0;
}

.c2 {
  --rows: 1;
  grid-row: span var(--rows);
}

.c2 > * {
  display: block;
  height: 100%;
}

.c1 {
  grid-template-rows: 1px;
}

@supports (width: 639px ) {
  .c0 {
    grid-template-columns: repeat( auto-fit, minmax(min(var(--minItemWidth),100%),1fr) );
  }
}

<div
  className="c0 c1"
  data-bedrock-layout-masonry-grid=""
>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
</div>
`;

exports[`MasonryGrid incorrect usage renders default with wrong gutter input 1`] = `
.c0 {
  box-sizing: border-box;
  --gutter: 0px;
  --minItemWidth: 639px;
  display: grid;
  gap: var(--gutter);
  grid-template-columns: repeat( auto-fit, minmax(min(639px,100%),1fr) );
}

.c0 > * {
  margin: 0;
}

.c2 {
  --rows: 1;
  grid-row: span var(--rows);
}

.c2 > * {
  display: block;
  height: 100%;
}

.c1 {
  grid-template-rows: 1px;
}

@supports (width: 639px ) {
  .c0 {
    grid-template-columns: repeat( auto-fit, minmax(min(var(--minItemWidth),100%),1fr) );
  }
}

<div
  className="c0 c1"
  data-bedrock-layout-masonry-grid=""
>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="c2"
    style={
      Object {
        "--rows": 1,
      }
    }
  >
    <div>
      1
    </div>
  </div>
</div>
`;
